gobblin-cluster/src/main/java/org/apache/gobblin/cluster/temporal/GobblinTemporalJobScheduler.java

```diff
@@ -0,0 +1,285 @@
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.gobblin.cluster.*;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.typesafe.config.Config;
+
+import org.apache.gobblin.annotation.Alpha;
+import org.apache.gobblin.cluster.event.CancelJobConfigArrivalEvent;
+import org.apache.gobblin.cluster.event.DeleteJobConfigArrivalEvent;
+import org.apache.gobblin.cluster.event.NewJobConfigArrivalEvent;
+import org.apache.gobblin.cluster.event.UpdateJobConfigArrivalEvent;
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.instrumented.Instrumented;
+import org.apache.gobblin.instrumented.StandardMetricsBridge;
+import org.apache.gobblin.metrics.MetricContext;
+import org.apache.gobblin.metrics.Tag;
+import org.apache.gobblin.runtime.JobException;
+import org.apache.gobblin.runtime.JobLauncher;
+import org.apache.gobblin.runtime.listeners.JobListener;
+import org.apache.gobblin.scheduler.JobScheduler;
+import org.apache.gobblin.scheduler.SchedulerService;
+import org.apache.gobblin.util.ConfigUtils;
+import org.apache.gobblin.util.PathUtils;
+import org.apache.gobblin.util.PropertiesUtils;
+
+
+/**
+ * An extension to {@link JobScheduler} that schedules and runs
+ * Gobblin jobs on Temporal.
+ *
+ * <p> If the job should be launched from the scheduler node,
+ * {@link GobblinTemporalJobLauncher} is invoked.
+ * TODO(yiyang): this file should be cleaned up with HelixJobScheduler.
+ */
+@Alpha
+public class GobblinTemporalJobScheduler extends JobScheduler implements StandardMetricsBridge {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GobblinTemporalJobScheduler.class);
+  private static final String COMMON_JOB_PROPS = "gobblin.common.job.props";
+
+  private final Properties commonJobProperties;
+  private final EventBus eventBus;
+  private final Path appWorkDir;
+  private final List<? extends Tag<?>> metadataTags;
+  private final ConcurrentHashMap<String, Boolean> jobRunningMap;
+  private final MetricContext metricContext;
+  final GobblinHelixJobSchedulerMetrics jobSchedulerMetrics;
+  final GobblinHelixJobLauncherMetrics launcherMetrics;
+  final GobblinHelixPlanningJobLauncherMetrics planningJobLauncherMetrics;
+  final HelixJobsMapping jobsMapping;
+  private boolean startServicesCompleted;
+
+  public GobblinTemporalJobScheduler(Config sysConfig,
+                                     EventBus eventBus,
+                                     Path appWorkDir, List<? extends Tag<?>> metadataTags,
+                                     SchedulerService schedulerService) throws Exception {
```

Comment:
line-continuation indentation should be four spaces.  see the `GobblinHelixJobLauncher` ctor as an example